### PR TITLE
Feature/mental health playback ab test

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

Test group `C` and `D` are not implemented yet because the design is not ready yet.

https://www.notion.so/joinzoe/017c3de591604a7b89d0b69e3c37b279?v=0133357df3ed42e1bd42c1e0d7afcf45&p=7c6ea57a7d4e4bb48dbe211ea96297bf

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Simulator Screen Shot - iPhone 11 - 2021-07-14 at 21 27 53

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

**Comment by juliatan** [comment]
Looks fine to me. Didn't see where GROUP B fit in though.

File: src/components/Modal.tsx [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import { SafeLayout } from '@covid/components';
 import * as React from 'react';
-import { Modal, ScrollView, StyleSheet } from 'react-native';
+import { Modal as RNModal, ScrollView, StyleSheet } from 'react-native';
```
Comment (bionara) [inline src/components/Modal.tsx:3]: @aikewoody what's the reason for the import as?

File: src/features/mental-health-playback/util.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+class Util {
+  determineTestGroupId(id: string): TTestGroupId {
+    // Example of execution:
+    // 1: 2982e1b0-aa5e-4c74-9c56-c059cdf4b294
+    // 2: 2982e1b0aa5e4c749c56c059cdf4b294
+    // 3: 5.517792395783495e+37
+    // 4: 5517792395783495e+37
+    // 5: 5517
+    // 6: 6
+    const remainder =
```
Comment (bionara) [inline src/features/mental-health-playback/util.ts:12]: I wonder if we even need to use the user id (minus letters!) for this vs just generating a random number?

**Review by bionara — APPROVED** [review]
Nice work @aikewoody !
Obviously AB testing introduces a lot of redundant code - I wonder if each time we do AB test if we should automatically create a cleanup ticket, and tag each code piece (for the AB test) with a // TODO: ticket-url for easier maintenance?

File: src/features/mental-health-playback/util.ts (hunk 2 above)
Comment (aikewoody) [reply src/features/mental-health-playback/util.ts:12]: By using this method it's deterministic (given a user we can always determine the group they belong to) and we don't need to story any additional data

**Comment by aikewoody** [comment]
> Looks fine to me. Didn't see where GROUP B fit in though.

@juliatan Group B is show the normal modal after the daily report, the following code in the file `src/features/thank-you/ThankYouUKScreen.tsx `:
```
if (testGroupId === 'GROUP_B') {
  dispatch(appActions.setModalMentalHealthPlaybackVisible(true));
  return;
}
```

**Pull request merged**